test/source/TestFunctional.cpp

```diff
@@ -1491,5 +1491,39 @@ struct TestInvokeResult
 };
 
 template struct eastl::invoke_result<decltype(&TestInvokeResult::f), TestInvokeResult, void>;
+
 static_assert(!eastl::is_invocable<decltype(&TestInvokeResult::f), TestInvokeResult, void>::value, "incorrect value for is_invocable");
+static_assert(!eastl::is_invocable<decltype(&TestInvokeResult::f), TestInvokeResult, int, int>::value, "incorrect value for is_invocable");
 static_assert(eastl::is_invocable<decltype(&TestInvokeResult::f), TestInvokeResult, int>::value, "incorrect value for is_invocable");
+
+static_assert(!eastl::is_invocable_r<int, decltype(&TestInvokeResult::f), TestInvokeResult, void>::value, "incorrect value for is_invocable_r");
+static_assert(!eastl::is_invocable_r<void, decltype(&TestInvokeResult::f), TestInvokeResult, int>::value, "incorrect value for is_invocable_r");
```

Comment:
I believe this goes against the spec and it's a bug in `eastl::is_invocable_r` (a different bug from the one this PR is trying to fix). From the docs on [std::is_invocable_r](https://en.cppreference.com/w/cpp/types/is_invocable)

>  ... If R is cv void, the result can be any type...

I've verified this by plugging these tests into compiler explorer [here](https://godbolt.org/z/sxMcGjbM4). 

I think the fix is to add a disjunction in `is_invocable_r_impl`, something like this:
```
	template <typename R, typename F, typename... Args>
	struct is_invocable_r_impl<R, F, void_t<typename invoke_result<F, Args...>::type>, Args...>
	    : public disjunction<is_convertible<typename invoke_result<F, Args...>::type, R>,
	                         is_same<typename remove_cv<R>::type, void>> {};
```
Whether you want to put that fix in as part of this PR or not is up to you. I think it'd be good to iron that out, but if you decide not to put the fix in then I'd ask to please correct the expected behavior in this test (i.e. remove the `!` from this static assert, and the similar ones below) and comment it out with a clarifying comment saying that the test is disabled due to a bug in `eastl::is_invocable_r` not conforming to the spec when `void` is passed as a return value.